File: SpendingControl/src/main/java/com/keita/spendingcontrol/model/dto/DailyAnalytic.java
package com.keita.spendingcontrol.model.dto;

import com.keita.spendingcontrol.model.enums.DegreeOfUseFullness;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DailyAnalytic implements Serializable {
    private Map<DegreeOfUseFullness,ArticleDetail> mostExpensiveArticlesByUseFullness,lessExpensiveArticlesByUseFullness;
    private ArticleDetail mostExpensiveArticle, lessExpensiveArticle;
    private Map<DegreeOfUseFullness,Float> totalByUseFullness;
    private Float totalPrice;
}

File: SpendingControlAng/src/app/model/daily-analytic.ts
import { Article } from "./article";

export class DailyAnalytic {
    id:number;
    mostExpensiveArticlesByUseFullness:Map<string,Article> = new Map();
    lessExpensiveArticlesByUseFullness:Map<string,Article> = new Map();
    mostExpensiveArticle: Article = new Article();
    lessExpensiveArticle: Article = new Article();
    totalByUseFullness:Map<string,number> = new Map();
    totalPrice:number;
}
